pkg/workflow_template.go

```diff
@@ -346,6 +369,28 @@ func (c *Client) GetWorkflowTemplate(namespace, uid string, version int32) (work
 	return
 }
 
+func (c *Client) GetWorkflowTemplateByName(namespace, name string, version int32) (workflowTemplate *WorkflowTemplate, err error) {
+	workflowTemplate, err = c.getWorkflowTemplateByName(namespace, name, version)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"Namespace":        namespace,
+			"WorkflowTemplate": workflowTemplate,
+			"Error":            err.Error(),
+		}).Error("Get Workflow Template failed.")
+		return nil, util.NewUserError(codes.Unknown, "Unknown error.")
+	}
+	if workflowTemplate == nil {
+		return nil, util.NewUserError(codes.NotFound, "Workflow template not found.")
+	}
+
+	return
+}
+
+func (c *Client) CloneWorkflowTemplate() (workflowTemplate *WorkflowTemplate, err error) {
```

Comment:
Is this supposed to do anything?

Reply:
Good catch. I ended up implementing the functionality in the Server piece instead of the Client.